src/realm/object-store/sync/sync_session.cpp

```diff
@@ -319,37 +328,41 @@ void SyncSession::download_fresh_realm()
     catch (std::exception const& e) {
         // Failed to open the fresh path after attempting to delete it, so we
         // just can't do automatic recovery.
-        handle_fresh_realm_downloaded(nullptr, std::string(e.what()));
+        handle_fresh_realm_downloaded(nullptr, std::string(e.what()), allowed_mode);
         return;
     }
 
     std::shared_ptr<SyncSession> sync_session;
-    {
-        util::CheckedLockGuard state_lock(m_state_mutex);
-        util::CheckedLockGuard config_lock(m_config_mutex);
-        SyncConfig config = m_config;
-        config.stop_policy = SyncSessionStopPolicy::Immediately;
-        config.client_resync_mode = ClientResyncMode::Manual;
-        sync_session = create(m_client, db, config, m_sync_manager);
+
+    util::CheckedLockGuard state_lock(m_state_mutex);
+    if (m_state != State::Active) {
+        return;
     }
+    util::CheckedLockGuard config_lock(m_config_mutex);
```

Comment:
do we want to hold the config_log while we're waiting for download completion?

Reply:
right, the config lock isn't needed.